packages/block-editor/src/components/link-control/index.js

```diff
@@ -306,27 +475,57 @@ function LinkControl( {
 			ref={ wrapperNode }
 			className="block-editor-link-control"
 		>
-			{ isEditingLink || ! value ? (
+			{ isResolvingLink && (
+				<div className="block-editor-link-control__loading">
+					<Spinner /> { __( 'Creating' ) }…
+				</div>
+			) }
+
+			{ ( isEditingLink || ! value ) && ! isResolvingLink && (
 				<LinkControlSearchInput
 					value={ inputValue }
 					onChange={ onInputChange }
 					onSelect={ ( suggestion ) => {
-						onChange( { ...value, ...suggestion } );
-						stopEditing();
+						if (
+							suggestion.type &&
+							CREATE_TYPE === suggestion.type
+						) {
+							cancelableOnCreate = makeCancelable(
+								handleOnCreate( inputValue )
+							)
+								.promise.then( () => stopEditing() )
+								.catch( ( error ) => {
+									if ( error.isCanceled ) {
+										return; // bail if canceled to avoid setting state
+									}
+
+									stopEditing();
+								} );
+						} else {
+							handleSelectSuggestion( suggestion, value );
+							// Must be called after handling to ensure focus is
+							// managed correctly.
+							stopEditing();
+						}
 					} }
 					renderSuggestions={ renderSearchResults }
 					fetchSuggestions={ getSearchHandler }
 					onReset={ resetInput }
 					showInitialSuggestions={ showInitialSuggestions }
+					errorMessage={ errorMessage }
 				/>
-			) : (
+			) }
+
+			{ value && ! isEditingLink && ! isResolvingLink && (
 				<Fragment>
-					<p
-						className="screen-reader-text"
-						id={ `current-link-label-${ instanceId }` }
-					>
-						{ __( 'Currently selected' ) }:
-					</p>
+					<VisuallyHidden>
+						<p
+							aria-label={ __( 'Currently selected' ) }
+							id={ `current-link-label-${ instanceId }` }
+						>
+							{ __( 'Currently selected' ) }:
+						</p>
+					</VisuallyHidden>
```

Comment:
Related to 4f87b156e9bdc4f4dab4ac8532b81a3ffd9838d8 and the discussion at https://github.com/WordPress/gutenberg/pull/19775#discussion_r374805886 :

I am conflicted on these changes:

- `VisuallyHidden` is the preferred alternative to `screen-reader-text` class, yes.
- The addition of `aria-label` to the `p` appears that it would be redundant, if it's the same as the text otherwise rendered as the sole child of the element. **Edit:** I just noticed the colon `:` in the child, so I could see an argument for why the `aria-label` would be there if the colon would otherwise be confusing in the context of screen readers and other assistive technology.
- Like with 4f87b156e9bdc4f4dab4ac8532b81a3ffd9838d8, `aria-labelledby` should be avoided when there is no visible label on the screen.

I stumbled upon this snippet:

>If the interface is such that it is not possible to have a visible label on the screen, authors **SHOULD** use `aria-label` and **SHOULD NOT** use `aria-labelledby`. 

https://www.w3.org/TR/wai-aria/#aria-labelledby

... Which seems to describe the implementation we have currently.

I have more questions about the labels more generally (e.g. how exactly we're expecting that they be announced). I realize, however, that this isn't strictly relevant for the purpose of your pull request. That said, this revision was included, so it caught my attention for review.

My recommendation would be one of:

- If it's not strictly relevant or necessary for the goals of your pull request, revert it to the original state and deal with the problems separately.
- If you choose to take on these issues here, then this may need some further revisions per the above comments. **Edit:** My earlier-stated misunderstanding about `aria-label` was the main thing that drew me to think that this wasn't strictly an improvement over the previous implementation. It may be "fine" to leave as-is if we can grant that, while not perfect, it's an improvement over the previous implementation.

Reply:
Ok this is getting confusing 😄 

`aria-labelledby` is only for when the referenced element (the one which "labels" the element on which the attribute is set) is visible.

Therefore you are correct and I have now removed the `aria-labelledby` on this.

`aria-label` is not supposed to be used to duplicate content already accessible within the content. I don't think the colon argument is solid and so I've also removed that. 

> If the interface is such that it is not possible to have a visible label on the screen, authors SHOULD use aria-label

I also noted this. Therefore I've retained the `aria-label` on the `div.block-editor-link-control__search-item` element as the label is not visible.

Reply:
I believe you could remove the entire `<VisuallyHidden>` section, as the `aria-label` is in place on the `div.block-editor-link-control__search-item`. 

The label itself is somewhat confusing to me. Is `Currently selected`  accurately describing the state of the `LinkControl`? 

As far as next steps, I think:
- remove the `<VisuallyHidden>Currently selected</VisuallyHidden>`
- leave as is for this PR
- ask for advice from an accessibility expert on the best label text to use (or if the implementation should be changed)

Reply:
Done